src/menu/index.tsx

```diff
@@ -224,7 +226,7 @@ export const Menu = factory(function({
 
 	const itemToScroll = icache.get('itemToScroll');
 	const menuHeight = icache.get('menuHeight');
-	const idBase = icache.getOrSet('idBase', uuid());
+	const idBase = widgetId || id;
```

Comment:
you should prefix the `id` with the widget name or something as it will just be an incrementing number like `6`, which isn't very unique in a page.